src/main/java/org/opensearch/knn/indices/Model.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.indices;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.opensearch.knn.index.SpaceType;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import java.util.Objects;
+
+import static org.opensearch.knn.index.KNNVectorFieldMapper.MAX_DIMENSION;
+
+public class Model {
+
+    final private KNNEngine knnEngine;
+    final private SpaceType spaceType;
+    final private int dimension;
+    final private byte[] modelBlob;
+
+    /**
+     * Constructor
+     *
+     * @param knnEngine engine model is built with
+     * @param spaceType space type model uses
+     * @param modelBlob binary representation of model template index
+     */
+    public Model(KNNEngine knnEngine, SpaceType spaceType, int dimension, byte[] modelBlob) {
+        this.knnEngine = Objects.requireNonNull(knnEngine, "knnEngine must not be null");
+        this.spaceType = Objects.requireNonNull(spaceType, "spaceType must not be null");
+        if (dimension <= 0 || dimension >= MAX_DIMENSION) {
+            throw new IllegalArgumentException("Dimension value must be greater than 0 and less than " + MAX_DIMENSION);
```

Comment:
Add given dimension to the message as well?

Reply:
Will add